algorithms/tests/snark/mimc.rs

```diff
@@ -172,8 +172,8 @@ fn test_mimc_groth_16() {
     // Create parameters for our circuit
     let params = {
         let c = MiMCDemo::<Fr> {
-            xl: None,
-            xr: None,
+            xl: Some(rng.gen()),
+            xr: Some(rng.gen()),
```

Comment:
These should not need to exist for the setup to run. For testing purposes, to ensure that is the case, it is best to leave it unassigned.

Reply:
If either `xl` or `xr` is `None`, `generate_constraints` will fail at the first call to `alloc`, crashing the test with `AssignmentMissing`.

An alternative approach could be to only execute the closures in `alloc` and `alloc_input` outside of `#[cfg(test)]` conditions, if that's preferable.